akka-docs/src/test/scala/docs/stream/IntegrationDocSpec.scala

```diff
@@ -130,6 +127,74 @@ object IntegrationDocSpec {
     }
   }
   //#ask-actor
+  class ActorRefWithBackPressure {
+    //#actorref-with-ack
+    val initMessage = "start"
+    val onCompleteMessage = "done"
+    val ackMessage = "ack"
+
+    implicit val system = ActorSystem("ActorWithBackPressure")
+    implicit val mat = ActorMaterializer()
+
+    class ActorWithBackPressure(ref: ActorRef) extends Actor {
+      def receive = {
+        case `initMessage` ⇒
+          sender() ! ackMessage
+          ref forward initMessage
+        case `onCompleteMessage` ⇒
+          ref forward onCompleteMessage
+        case msg: Int ⇒
+          sender() ! ackMessage
+          ref forward msg
+      }
+    }
+    class MyActor extends Actor {
+      def receive: Receive = {
+        case `initMessage`       ⇒ println("start")
+        case `onCompleteMessage` ⇒ println("done!")
+        case msg: Int            ⇒ println(s"processing $msg")
+      }
+    }
+
+    val actor = system.actorOf(
+      Props(classOf[ActorWithBackPressure], system.actorOf(Props[MyActor])))
+
+    Source(List(1, 2, 3)).runWith(
+      Sink.actorRefWithAck(actor, initMessage, ackMessage, onCompleteMessage))
+
+    //#actorref-with-ack
+
+  }
+
+  class PublishToSourceQueue {
+    //#source-queue
+    implicit val system = ActorSystem("SourceQueue")
+    implicit val materializer = ActorMaterializer()
+
+    val queue = Source.queue[Int](bufferSize = 2, OverflowStrategy.backpressure)
+      .groupedWithin(2, 2.seconds)
+      .mapAsyncUnordered(2) { elem =>
+        Future {
+          println(s"${Thread.currentThread().getName} adding delay, $elem")
+          Thread.sleep(1000L)
+          elem
+        }(scala.concurrent.ExecutionContext.global)
+      }.to(Sink.ignore)
+      .run
+
+    new Thread {
+      () =>
+        {
+          while (true) {
+            val offerResult: Future[QueueOfferResult] = queue.offer(1)
+            Await.ready(offerResult, 10.seconds)
+            println(s"${Thread.currentThread().getName} Emitted 1 $offerResult")
+          }
+        }
+    }.start()
```

Comment:
This code contains so many anti-patterns and even if we know that it's just an example that illustrates how it works I'm afraid that people will just copy-paste and put into production without understanding.

I suggest an example that has an Actor with a `Source.queue`. That actor offer to the queue when it receives a message, and then use `pipeTo` to reply to the original sender when the `Future` is completed.